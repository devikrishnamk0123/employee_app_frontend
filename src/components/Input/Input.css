label{
    font-family: Nunito;
    font-weight: SemiBold;
    font-size: 13px;
    color: #3E3E3E;
    margin-bottom: -20px;
    /* font-size: 13px; */
}



input{
    width: 460px;
    height: 50px;
    border: 1px solid #D9D9D9;
    border-radius: 5px;
    margin-top: 30px;
    padding-left: 5px;
    font: normal SemiBold 600 13px Nunito;
    color: #000000;
    background: #FAFAFA 0% 0% no-repeat padding-box;
}

.loginform{
    display: flex;
    flex-direction: column;
}