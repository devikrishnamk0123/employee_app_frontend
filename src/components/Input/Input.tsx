import React from 'react';
import './Input.css';

type InputPropType = {
  label?: string;
  value?: string;
  placeholder: string;
  type: 'text' | 'password';
  onChange?: (value: any) => void;
};

const Input: React.FC<InputPropType> = (props) => {
  return (
    <div className='loginform'>
      {props.label && <label>{props.label}</label>}
      <input
        type={props.type}
        value={props.value}
        placeholder={props.placeholder}
        onChange={(e) => props.onChange(e.target.value)}
      />
    </div>
  );
};

export default Input;
