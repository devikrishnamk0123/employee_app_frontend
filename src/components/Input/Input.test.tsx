import Input, { InputPropType } from './Input';
import { fireEvent, render, screen } from '@testing-library/react';

describe('If Input works properly', () => {
  test('If placeholder rendered properly', () => {
    const InputProps: InputPropType = {
      type: 'text',
      placeholder: 'default placeholder'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element.getAttribute('placeholder')).toBe('default placeholder');
  });

  test('If type:text rendered properly', () => {
    const InputProps: InputPropType = {
      type: 'text',
      placeholder: 'hello'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element.getAttribute('type')).toBe('text');
    expect(element).toMatchSnapshot();
  });

  test('If value rendered properly', () => {
    const InputProps: InputPropType = {
      type: 'text',
      placeholder: 'hello',
      value: 'hello world'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element.getAttribute('value')).toBe('hello world');
    expect(element).toMatchSnapshot();
  });

  test('If onChange triggered properly', () => {
    const onChange = jest.fn();
    const InputProps: InputPropType = {
      type: 'text',
      placeholder: 'hello',
      value: 'hello world',
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    fireEvent.change(element);

    expect(onChange).toBeCalledTimes(1);
  });
});
