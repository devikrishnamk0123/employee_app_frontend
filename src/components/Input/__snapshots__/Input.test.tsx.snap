// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Input works properly If type:text rendered properly 1`] = `
<input
  data-testid="input-test"
  placeholder="hello"
  type="text"
  value=""
/>
`;

exports[`If Input works properly If value rendered properly 1`] = `
<input
  data-testid="input-test"
  placeholder="hello"
  type="text"
  value="hello world"
/>
`;
